src/scripts/main.js

```diff
@@ -1,3 +1,254 @@
+    field[row] = moving(field[row].reverse()).reverse();
+  }
+};
+
+const moveUp = function() {
+  for (let cell = 0; cell < field.length; cell++) {
+    let newRow = [
+      field[0][cell],
+      field[1][cell],
+      field[2][cell],
+      field[3][cell],
+    ];
+
+    newRow = moving(newRow);
+    field[0][cell] = newRow[0];
+    field[1][cell] = newRow[1];
+    field[2][cell] = newRow[2];
+    field[3][cell] = newRow[3];
+  }
+};
+
+const moveDown = function() {
+  for (let cell = 0; cell < field.length; cell++) {
+    let newRow = [
+      field[3][cell],
+      field[2][cell],
+      field[1][cell],
+      field[0][cell],
+    ];
+
+    newRow = moving(newRow);
+    field[0][cell] = newRow[3];
+    field[1][cell] = newRow[2];
+    field[2][cell] = newRow[1];
+    field[3][cell] = newRow[0];
+  }
+};
+
+const winCheck = function() {
+  for (let row = 0; row < field.length; row++) {
+    for (let cell = 0; cell < field[row].length; cell++) {
+      if (field[row][cell] === 2024) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+};
+
+const loseCheck = function() {
+  for (let row = 0; row < field.length; row++) {
+    for (let cell = 0; cell < field[row].length; cell++) {
+      if (field[row][cell] === 0) {
+        return false;
+      };
+
+      let checkUp, checkDown, checkRight, checkLeft;
+
+      if (field[row + 1]) {
+        checkUp = field[row][cell] === field[row + 1][cell];
+      }
+
+      if (field[row - 1]) {
+        checkDown = field[row][cell] === field[row - 1][cell];
+      }
+
+      if (field[row][cell + 1]) {
+        checkRight = field[row][cell] === field[row][cell + 1];
+      }
+
+      if (field[row][cell - 1]) {
+        checkLeft = field[row][cell] === field[row][cell - 1];
+      }
+
+      if (checkUp || checkDown || checkRight || checkLeft) {
+        return false;
+      }
+    }
+  }
+
+  return true;
+};
+// #endregion
+// #region interactivePage
+const page = document.documentElement;
+const score = document.querySelector('.game-score');
+const button = document.querySelector('.button');
+const table = document.querySelector('.game-field');
+const messageStart = document.querySelector('.message-start');
+const messageLose = document.querySelector('.message-lose');
+const messageWin = document.querySelector('.message-win');
+
+let field;
+let scoreCounter = 0;
+
+button.addEventListener('click', () => {
+  if (button.classList.contains('start')) {
+    button.classList.remove('start');
+    button.classList.add('restart');
+    button.textContent = 'Restart';
+    messageStart.classList.add('hidden');
+  }
+
+  messageLose.classList.add('hidden');
+  messageWin.classList.add('hidden');
+  button.classList.remove('restart.active');
+
+  field = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+
+  randomNew();
+  randomNew();
+});
+
+page.addEventListener('keydown', (eve) => {
+  eve.preventDefault();
+
+  switch (eve.key) {
```

Comment:
looks like some actions in your cases are the same. let's get rid of code duplication

Reply:
Thanks for your rewiew!
All done, except combining moveUp, moveDown, moveRight and moveLeft. I have no idea how to do it...